docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,312 @@
+
+```html
+<div>This is valid HTML &amp; JSX at the same time.</div>
+```
+
+JSX removes whitespace at the beginning and ending of a line. It also removes blank lines. So these all render to the same thing:
+
+```js
+<div>Hello</div>
+
+<div>Hello
+</div>
+
+<div>
+  Hello
+</div>
+
+<div>
+
+  Hello
+
+</div>
+```
+
+### JSX Children
+
+You can provide more JSX as the children. This is useful for displaying nested components:
+
+```js
+<MyContainer>
+  <MyFirstComponent />
+  <MySecondComponent />
+</MyContainer>
+```
+
+You can mix together different types of children, so you can use string literals together with JSX children. This is another way in which JSX is like HTML, so that this is both valid JSX and valid HTML:
+
+```html
+<div>
+  Here is a list:
+  <ul>
+    <li>Item 1</li>
+    <li>Item 2</li>
+  </ul>
+</div>
+```
+
+A React component can't return multiple React elements, but a single JSX expression can have multiple children, so if you want a component to render multiple things you can wrap it in a `div` like this.
+
+### JavaScript Expressions
+
+You can pass any JavaScript expression as children, by enclosing it within `{}`. For example, these expressions are equivalent:
+
+```js
+<MyComponent>foo</MyComponent>
+
+<MyComponent>{"foo"}</MyComponent>
+```
+
+This is often useful for rendering a list of JSX expressions of arbitrary length. For example, this renders an HTML list:
+
+```js
+function renderItem(message) {
+  return <li key={message}>{message}</li>;
+}
+
+function renderTodoList() {
+  var todos = ['finish doc', 'submit pr', 'nag dan to review'];
+  return (
+    <ul>
+      {todos.map(renderItem)}
+    </ul>
+  );
+}
+```
+
+JavaScript expressions can be mixed with other types of children. This is often useful in lieu of string templates:
+
+```js
+function Hello(props) {
+  return <div>Hello {props.addressee}!</div>;
+}
+```
+
+Normally, JavaScript expressions inserted in JSX will evaluate to a string, a React element, or a list of those things. However, `props.children` works just like any other prop in that it can pass any sort of data, not just the sorts that React knows how to render. For example, if you have a custom component, you could have it take a callback as `props.children`:
+
+```js
+// Repeats a component numTimes
+function Repeat(props) {
+  // The numbers 0 .. (numTimes - 1)
+  var range = [...Array(props.numTimes).keys()];
+  return <div>{range.map(props.children)}</div>
+}
+
+function ListOfTenThings() {
+  return (
+    <Repeat numTimes={10}>
+      {(index) => <div key={index}>This is item {index} in the list</div>}
+    </Repeat>
+  );
+}
+```
+
+This usage is not common, but it works if you want to stretch what JSX is capable of.
+
+`false` or `null` are valid children. They simply don't render. These JSX expressions will all render to the same thing:
```

Comment:
It would be helpful to explain why this is useful `<div>{showHeader && <Header />}<Content /></div>`.

P.S. true also is ignored. This is not JSX but is React DOM and might not belong here.